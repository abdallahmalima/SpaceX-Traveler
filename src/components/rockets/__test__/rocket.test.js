import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Rocket from '../rocket';
import store from '../../../redux/configureStore';

describe('Test Rocket Component', () => {
  it('check if component component match the snapshot', () => {
    const rocket = {
      id: 1,
      rocketName: 'Falcon 1',
      description: 'The Falcon 1 was an expendable launch system privately developed',
      flickrImage: 'https://imgur.com/DaCfMsj.jpg',
      reserved: true,
    };

    const rocketRender = render(
      <Provider store={store}>
        <Rocket rocket={rocket} />
      </Provider>,
    );

    expect(rocketRender).toMatchSnapshot();
  });

  it('check if the barge is shown when a rocket is reserved and check if has button with text Cancel Reservation', () => {
    const rocket = {
      id: 1,
      rocketName: 'Falcon 1',
      description: 'The Falcon 1 was an expendable launch system privately developed',
      flickrImage: 'https://imgur.com/DaCfMsj.jpg',
      reserved: true,
    };

    const { getByText } = render(
      <Provider store={store}>
        <Rocket rocket={rocket} />
      </Provider>,
    );
    const button = getByText(/Cancel Reservation/i, { selector: 'button' });

    expect(getByText(/Reserved/i)).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });

  it('check if button is shown with Reserve Rocket text when reserved:false', () => {
    const rocket = {
      id: 1,
      rocketName: 'Falcon 1',
      description: 'The Falcon 1 was an expendable launch system privately developed',
      flickrImage: 'https://imgur.com/DaCfMsj.jpg',
      reserved: false,
    };

    const { getByText } = render(
      <Provider store={store}>
        <Rocket rocket={rocket} />
      </Provider>,
    );

    const button = getByText(/Reserve Rocket/i, { selector: 'button' });
    expect(button).toBeInTheDocument();
  });
});
