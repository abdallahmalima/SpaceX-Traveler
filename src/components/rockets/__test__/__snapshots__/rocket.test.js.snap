// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rocket Component check if component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="rocketCard"
      >
        <img
          alt="Rocket"
          class="img"
          src="https://imgur.com/DaCfMsj.jpg"
        />
        <div
          class="details"
        >
          <h4>
            Falcon 1
          </h4>
          <p>
            <span
              class="reserved"
            >
              Reserved
            </span>
            The Falcon 1 was an expendable launch system privately developed
          </p>
          <button
            class="btnCanceled"
            type="button"
          >
            Cancel Reservation
          </button>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="rocketCard"
    >
      <img
        alt="Rocket"
        class="img"
        src="https://imgur.com/DaCfMsj.jpg"
      />
      <div
        class="details"
      >
        <h4>
          Falcon 1
        </h4>
        <p>
          <span
            class="reserved"
          >
            Reserved
          </span>
          The Falcon 1 was an expendable launch system privately developed
        </p>
        <button
          class="btnCanceled"
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Rocket Component check if component component match the snap 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="rocketCard"
      >
        <img
          alt="Rocket"
          class="img"
          src="https://imgur.com/DaCfMsj.jpg"
        />
        <div
          class="details"
        >
          <h4>
            Falcon 1
          </h4>
          <p>
            <span
              class="reserved"
            >
              Reserved
            </span>
            The Falcon 1 was an expendable launch system privately developed
          </p>
          <button
            class="btnCanceled"
            type="button"
          >
            Cancel Reservation
          </button>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="rocketCard"
    >
      <img
        alt="Rocket"
        class="img"
        src="https://imgur.com/DaCfMsj.jpg"
      />
      <div
        class="details"
      >
        <h4>
          Falcon 1
        </h4>
        <p>
          <span
            class="reserved"
          >
            Reserved
          </span>
          The Falcon 1 was an expendable launch system privately developed
        </p>
        <button
          class="btnCanceled"
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Rocket Component check if component component match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="rocketCard"
      >
        <img
          alt="Rocket"
          class="img"
          src="https://imgur.com/DaCfMsj.jpg"
        />
        <div
          class="details"
        >
          <h4>
            Falcon 1
          </h4>
          <p>
            <span
              class="reserved"
            >
              Reserved
            </span>
            The Falcon 1 was an expendable launch system privately developed
          </p>
          <button
            class="btnCanceled"
            type="button"
          >
            Cancel Reservation
          </button>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="rocketCard"
    >
      <img
        alt="Rocket"
        class="img"
        src="https://imgur.com/DaCfMsj.jpg"
      />
      <div
        class="details"
      >
        <h4>
          Falcon 1
        </h4>
        <p>
          <span
            class="reserved"
          >
            Reserved
          </span>
          The Falcon 1 was an expendable launch system privately developed
        </p>
        <button
          class="btnCanceled"
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Rocket Component check if h 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="rocketCard"
      >
        <img
          alt="Rocket"
          class="img"
          src="https://imgur.com/DaCfMsj.jpg"
        />
        <div
          class="details"
        >
          <h4>
            Falcon 1
          </h4>
          <p>
            <span
              class="reserved"
            >
              Reserved
            </span>
            The Falcon 1 was an expendable launch system privately developed
          </p>
          <button
            class="btnCanceled"
            type="button"
          >
            Cancel Reservation
          </button>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="rocketCard"
    >
      <img
        alt="Rocket"
        class="img"
        src="https://imgur.com/DaCfMsj.jpg"
      />
      <div
        class="details"
      >
        <h4>
          Falcon 1
        </h4>
        <p>
          <span
            class="reserved"
          >
            Reserved
          </span>
          The Falcon 1 was an expendable launch system privately developed
        </p>
        <button
          class="btnCanceled"
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Rocket Component check if home page heading text render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="rocketCard"
      >
        <img
          alt="Rocket"
          class="img"
          src="https://imgur.com/DaCfMsj.jpg"
        />
        <div
          class="details"
        >
          <h4>
            Falcon 1
          </h4>
          <p>
            <span
              class="reserved"
            >
              Reserved
            </span>
            The Falcon 1 was an expendable launch system privately developed
          </p>
          <button
            class="btnCanceled"
            type="button"
          >
            Cancel Reservation
          </button>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="rocketCard"
    >
      <img
        alt="Rocket"
        class="img"
        src="https://imgur.com/DaCfMsj.jpg"
      />
      <div
        class="details"
      >
        <h4>
          Falcon 1
        </h4>
        <p>
          <span
            class="reserved"
          >
            Reserved
          </span>
          The Falcon 1 was an expendable launch system privately developed
        </p>
        <button
          class="btnCanceled"
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
